From 95381d4369d4502eab839dc69149e9f5b593525b Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Wed, 17 Dec 2014 21:38:57 -0500
Subject: [drm] radeon: don't reset dma on r6xx-evergreen init

Message-id: <1418852380-13061-47-git-send-email-rclark@redhat.com>
Patchwork-id: 102179
O-Subject: [RHEL7 drm 46/89] drm/radeon: don't reset dma on r6xx-evergreen init
Bugzilla: 1173317
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1173317
Upstream: since v3.16.4

commit b2e8e9cb9979a04703963fd96999e5d6b7380d2d

Author:     Alex Deucher <alexander.deucher@amd.com>
AuthorDate: Thu Sep 18 10:23:04 2014 -0400
Commit:     Rob Clark <rclark@redhat.com>
CommitDate: Wed Dec 10 17:19:59 2014 -0500

    drm/radeon: don't reset dma on r6xx-evergreen init

    commit c1789a2e66a4209fe5035eca11fdd729b2ffdd82 upstream.

    Otherwise we may lose the DMA golden settings which can
    lead to hangs, etc.

    Reviewed-by: Christian Koenig <christian.koenig@amd.com>
    Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Rob Clark <rclark@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/radeon/r600_dma.c b/drivers/gpu/drm/radeon/r600_dma.c
index 4969cef..b766e05 100644
--- a/drivers/gpu/drm/radeon/r600_dma.c
+++ b/drivers/gpu/drm/radeon/r600_dma.c
@@ -124,15 +124,6 @@ int r600_dma_resume(struct radeon_device *rdev)
  u32 rb_bufsz;
  int r;
 
- /* Reset dma */
- if (rdev->family >= CHIP_RV770)
-  WREG32(SRBM_SOFT_RESET, RV770_SOFT_RESET_DMA);
- else
-  WREG32(SRBM_SOFT_RESET, SOFT_RESET_DMA);
- RREG32(SRBM_SOFT_RESET);
- udelay(50);
- WREG32(SRBM_SOFT_RESET, 0);
-
  WREG32(DMA_SEM_INCOMPLETE_TIMER_CNTL, 0);
  WREG32(DMA_SEM_WAIT_FAIL_TIMER_CNTL, 0);
 
-- 
1.7.1